lib/http_ntlm.c

```diff
@@ -184,50 +197,54 @@ CURLcode Curl_output_ntlm(struct Curl_easy *data, bool proxy)
 #endif
 #endif
 
+  Curl_bufref_init(&ntlmmsg);
   switch(*state) {
   case NTLMSTATE_TYPE1:
   default: /* for the weird cases we (re)start here */
     /* Create a type-1 message */
     result = Curl_auth_create_ntlm_type1_message(data, userp, passwdp,
                                                  service, hostname,
-                                                 ntlm, &base64,
-                                                 &len);
-    if(result)
-      return result;
-
-    if(base64) {
-      free(*allocuserpwd);
-      *allocuserpwd = aprintf("%sAuthorization: NTLM %s\r\n",
-                              proxy ? "Proxy-" : "",
-                              base64);
-      free(base64);
-      if(!*allocuserpwd)
-        return CURLE_OUT_OF_MEMORY;
-
-      DEBUG_OUT(fprintf(stderr, "**** Header %s\n ", *allocuserpwd));
+                                                 ntlm, &ntlmmsg);
+    if(!result && Curl_bufref_len(&ntlmmsg)) {
```

Comment:
Is this actually expected to ever get an `ntlmmsg` back that is zero length when not an error? If not, then maybe a DEBUGASSERT for that before this check just to make it utterly obvious that this check is only for extra precaution?

Reply:
Yes, I think we can do so. The length test is just a translation of the previous code that already did the same check (on `base64`).

Reply:
> ... a DEBUGASSERT for that before this check ...

better do it **in** the `if`!